### DISCUSSION TITLE
How can I create a modal with a custom width?

### DISCUSSION THREAD

**Discussion by TheFe91** [rsuite/rsuite#4075] [Q&A] [answered]
I'm in the situation where the "lg" modal is too small for my needings but I don't want to switch to the "full" version. Is there a way I can pass the dimension or maybe a string percentage representing a portion of the total viewport width (e.g. "70%") ?

**Comment by simonguo** [comment]
Directly set a 70% for size

```
<Modal size="70%" >
```

**Reply by TheFe91** [reply]
This is not working, it is something that I already tried. Every time I set any string that is different from `xs` | `sm` | `md` | `lg`, the effect is that the modal takes the 100% of the width. In other words, entering `full`, `70%`, `arandomvalue`, `I am the best programmer in the world` everything behaves as `full`. IDK if this is a bug or a wanted behavior but I find it very limiting 🙁

**Reply by simonguo** [reply]
I tried in the document example, it works fine. I can't reproduce the problem you mentioned. What version of rsuite are you using? Or provide a codesandbox example for me. 
![image](https://github.com/user-attachments/assets/9b9e745e-9454-4ca7-9975-f786048ab5ac)

**Reply by TheFe91** [reply, marked as answer]
F**k, you're right, by using directly the Modal from `rsuite` it works nice...I wrapped the modal around a component of mine in order to customize it for my needings inside my design system and I must have broken something!
Ok, I'll fix it, thanks
